pkgs/ffigen/lib/src/config_provider/overrideable_utils.dart

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2025, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// This file contains utils that are intended to be overridden by downstream
+// clones. They're gathered into one file to make it easy to swap them out.
+
+import 'dart:io';
+
+import 'package:path/path.dart' as p;
+
+// Replaces the path separators according to current platform.
```

Comment:
What about `interface class PathHandler` and `class DefaultPathHandler` in the `main` function? That way it's a single constructor call in the main function. And we we use a different main function downstream there is no need for a git patch at all. (This is how the patching for native assets in flutter tools in g3 works.)

Reply:
I'm pretty sure ffigen uses the same main function in g3 as it does externally, so I'd have to write a patch file either way

Reply:
Done